src/cow-react/modules/limitOrders/containers/LimitOrdersReceiptModal/index.tsx

```diff
@@ -0,0 +1,80 @@
+import { ReceiptModal as ReceiptModalPure } from '@cow/modules/limitOrders/pure/ReceiptModal'
+import { useAtomValue } from 'jotai/utils'
+import { receiptAtom } from '@cow/modules/limitOrders/state/limitOrdersReceiptAtom'
+import { useCloseReceiptModal } from './hooks'
+import { useWeb3React } from '@web3-react/core'
+import { supportedChainId } from '@src/custom/utils/supportedChainId'
+import { CurrencyAmount, Fraction, Token } from '@uniswap/sdk-core'
+import JSBI from 'jsbi'
+
+function calculatePrice({
+  buyAmount,
+  sellAmount,
+  inputToken,
+  outputToken,
+}: {
+  buyAmount: JSBI | undefined
+  sellAmount: JSBI | undefined
+  inputToken: Token
+  outputToken: Token
+}) {
+  const isZero = (x: JSBI) => JSBI.equal(x, JSBI.BigInt(0))
+
+  if (!buyAmount || !sellAmount || !inputToken || !outputToken || isZero(buyAmount) || isZero(sellAmount)) {
+    return null
+  }
+
+  const adjustedBuy = adjustDecimals(buyAmount, inputToken.decimals)
+  const adjustedSell = adjustDecimals(sellAmount, outputToken.decimals)
+  return new Fraction(adjustedBuy, adjustedSell)
+}
+
+const adjustDecimals = (amount: string | JSBI, decimals: number) => {
```

Comment:
It's not clear what is this function given the name.

It's a function that calculate the `atoms` given the `decimal` amount? I think we need a better name, and some JS docs. 

Also, we should move to the utils, and make some tests to assert that it does exactly what you expect. 

Lastly, I would be surprised if we don't have a similar util somewhere

Reply:
Removed this function since its not needed anymore